### ISSUE TITLE
MueLu:  Amesos2 coarse level solver with Epetra support

### ISSUE THREAD

**Issue by maxfirmbach** [trilinos/Trilinos#11581]
## Question

Hello everyone,

in the context of switching from `Epetra` to `Tpetra` we are trying to switch the `Amesos` dependency to `Amesos2` in our in-house code.
There are mainly two use cases:
- directly interface with the direct solver
- use the direct method as coarse level solver in `MueLu`

We still use `Epetra` as backend for both packages. The problem I might/will run into is the following:
In `MueLu` it states I can only use the `Amesos2` interface with `Tpetra` (which we can't use / switch to right now), which can be problematic for us, as we would need to have both `Amesos` and `Amesos2` configured and in use:
- `Amesos` as interface for `MueLu` to coarse level direct solvers
- `Amesos2` for interfacing to direct solvers in our in-house code

Having newer versions of `Superlu_dist` and `Umfpack` installed, they might not be supported by both anymore / or some other dependency problems etc. occur.

Long story short, is there a reason to only allow `Amesos2` with `Tpetra` in `MueLu`? Having the flexibility to be able to use it with `Epetra` would be quite nice for slowly and step by step (package by package) switching to the new stack and still be able to use all features.

Best regards,
Max

@muelu and might also be interesting for @mayrmt @vryy

**Comment by cgcgcg** [comment]
I don't really know why this was never done. My best guess is that because there is no incredibly large performance difference between Amesos and Amesos2, it never mattered for us to run the combination of Epetra, MueLu and Amesos2.
